lightning/src/routing/network_graph.rs

```diff
@@ -540,26 +541,46 @@ impl NetworkGraph {
 		None
 	}
 
-	/// Scores each channel on the route by updating its channel_score instance within its
-	/// respective DirectionalChannelInfo.
-	pub fn score_channels_on_route(&mut self, route: Route, payment_succeeded: bool) -> Result<bool, LightningError> {
+	/// Increments payment_sent_score for appropriate DirectionalChannelInfos on the network graph.
+	pub fn score_payment_sent_for_route(&mut self, route: Route) -> Result<bool, LightningError> {
 		let paths = route.paths;
 		for path in paths {
-			for route_hop in path {
-				let short_channel_id = route_hop.short_channel_id;
-				let channel = self.channels.get_mut(&short_channel_id).unwrap();
-				let directional_channel_info = if route_hop.pubkey == channel.node_one {
-					channel.one_to_two.as_mut().unwrap()
-				} else {
-					channel.two_to_one.as_mut().unwrap()
-				};
-				let channel_score = &mut directional_channel_info.channel_score;
-				channel_score.payment_sent_score += 1;
+			self.increment_payment_sent_scores(path);
+		}
+		Ok(true)
+	}
+
+	/// Increments payment_failed_score for approriate DirectionalChannelInfos on the network
+	/// graph.
+	pub fn score_payment_failed_for_route(&mut self, route: Route, results: Vec<Result<(), APIError>>) -> Result<bool, LightningError> {
```

Comment:
Okay I think your approach is trying to encompass non-MPP/MPP payments separately  ? It would be better to have one function for both, if `results.len() == 0`, it may means every channel from every path of the MPP has succeeded and score them all higher. If not, iterate through Result and score accordingly ?

Reply:
Previously, I was not accounting for the MPP case. I have changed the `result` parameter to be called `results` of type `Vec<PublicKey>` that represents all the nodes that failed. If a node is in the results vector and somewhere in the `route`, then it will get dinged on its failed payment score.